### ISSUE TITLE
ch_sendexpr should support sending id as string when mode is lsp

### ISSUE THREAD

**Issue by prabirshrestha** [vim/vim#14091]
### Steps to reproduce

Use `pylsp-all` language server with work done progress enabled to see the error.

Since the server returns the id as string, the client needs to send it as string. The following command fails as id as string is not supported by vim https://github.com/vim/vim/blob/d086b8f646a67f6b16c46061ce773de1011b8ec7/src/channel.c#L4509C1-L4515C3

```vim
call ch_sendexpr(l:channel_id, {"id":"dea66433-b2b7-4e22-af71-04e4f608164f","jsonrpc":"2.0","result":null})
```

This was first reported at https://github.com/prabirshrestha/vim-lsp/issues/1497.

Since Rust returns id as integer, there is no issue when running rust language server.

### Expected behaviour

Shouldn't throw error.

### Version of Vim

9.1.100

### Environment

OSX

**Comment by Shane-XB-Qian** [comment]
tho spec was saying the id of resp could be string/int/null, but even it was a string, mostly it was a string like int.
// almost all was like that, tho pylsp seems was the special or a bit different, :smile:
in yegappan's lsp, i limited it (the req id) to int or "string like int" only.

**Comment by yegappan** [comment]
Currently the Vim channel code supports only using number as a message identifier.  A reply message is matched with a request message using a simple number comparison.  To support a string as an identifier will need a non-trivial change.

**Comment by castedo** [comment]
For anybody waiting on this bug/enhancement, consider the following three line function using `ch_sendraw`:
https://github.com/yegappan/lsp/blob/34e071c3e860f5567b02122ead20676aa65f0d48/autoload/lsp/lspserver.vim#L362
that has just been merged into yegappan/lsp (https://github.com/yegappan/lsp/pull/768).

I agree that it would be nice if `ch_sendexpr` can handle string ids.

That said, if the objective is a vim LSP *client* replying with LSP responses with string ids, then they do not need the extra RPC request/response callback logic of `ch_sendexpr`. They only need to send an LSP *response* message (not LSP *request* message). And as demonstrated by the above 3 line function, JSONRPC response serialization is relatively simple and requires no callback tracking.

**Comment by castedo** [comment]
It might be worth distinguishing between `ch_sendexpr` called with vs without a `callback` item in the `options` argument.

It might be an easy fix to allow string id when there is no `callback` but otherwise error out. It makes documentation a little bit more complicated, but probably is fixing the main issue people are having.

I doubt it is very important to be able to call `ch_sendexpr` with a `callback` *and* intentionally set a string id to track the pending request/response call/callback interaction.
